### PR TITLE
DAOS-11604 control,mgmt,bio: Add dmg functionality for VMD LED manage…

### PR DESCRIPTION
**Pull request by tanabarr**
…ment (#10339)

New dmg functionality for VMD LED management:

$ dmg storage led identify [--reset]
$ dmg storage led check

--reset: Reset the status LED on a VMD SSD back to the previous state (OFF/normal or ON/faulty) after issuing an identify command. DEVICES: Comma-separated list of device-UUIDs or VMD backing PCI address

led-identify will set LED state to QUICK_BLINK or reset to the previous state. led-check will display the current state of the status LED on a VMD backing SSD.

LED state is now displayed in dmg storage query list-devices, dmg storage set nvme-faulty and dmg storage nvme replace cmd output. dmg storage identify vmd command has been removed. Configurable timeout parameter to be added in a subsequent patch, default timeout changed from one to two minutes.

Required-githooks: true

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>
Co-authored-by: Sydney Vanda <sydney.m.vanda@intel.com>

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficent testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by jolivier23 — DISMISSED** [review]

File: src/mgmt/srv_query.c [hunk 1]
```diff
@@ -652,36 +596,89 @@ ds_mgmt_dev_set_faulty(uuid_t dev_uuid, Ctl__DevStateResp *resp)
...
-	strncpy(resp->dev_state, state_str, buflen);
+	uuid_unparse_lower(dev_uuid, resp->device->uuid);
+
+	uuid_copy(led_info.dev_uuid, dev_uuid);
+	led_info.action = CTL__LED_ACTION__SET;
+	led_state = CTL__LED_STATE__ON;
+	led_info.state = &led_state;
+
+	/* Set the VMD LED to FAULTY state on init xstream */
+	rc = dss_ult_execute(bio_storage_dev_manage_led, &led_info, NULL, NULL, DSS_XS_VOS, 0, 0);
```
Comment (wangshilong) [inline src/mgmt/srv_query.c:624]: This should be DSS_XS_SYS

File: src/mgmt/srv_query.c [hunk 2]
```diff
@@ -652,36 +596,89 @@ ds_mgmt_dev_set_faulty(uuid_t dev_uuid, Ctl__DevStateResp *resp)
...
+	strncpy(resp->device->tr_addr, req->ids, ADDR_STR_MAX_LEN + 1);
+
+	/* tr_addr will be used if set and get populated if not */
+	led_info.tr_addr = resp->device->tr_addr;
+	led_info.action = req->led_action;
+	led_state = req->led_state;
+	led_info.state = &led_state;
+
+	/* Manage the VMD LED state on init xstream */
+	rc = dss_ult_execute(bio_storage_dev_manage_led, &led_info, NULL, NULL, DSS_XS_VOS, 0, 0);
```
Comment (wangshilong) [inline src/mgmt/srv_query.c:677]: DSS_XS_SYS

Comment (tanabarr) [reply src/mgmt/srv_query.c:677]: thanks, good catch

File: src/mgmt/srv_query.c (hunk 1 above)
Comment (tanabarr) [reply src/mgmt/srv_query.c:624]: thanks, good catch

Comment (tanabarr) [reply src/mgmt/srv_query.c]: done

File: src/mgmt/srv_query.c (hunk 2 above)
Comment (tanabarr) [reply src/mgmt/srv_query.c]: done

**Review by wangshilong — DISMISSED** [review]

**Comment by tanabarr** [comment]
No test tags applied as the PR is wide-ranging across control-plane and engine.

**Comment by jolivier23** [comment]
I think you should add your other patch to this backport so we don't break the branch

**Comment by tanabarr** [comment]
> I think you should add your other patch to this backport so we don't break the branch

yes indeed, done

**Review by jolivier23 — APPROVED** [review]

**Review by NiuYawei — APPROVED** [review]

**Review by knard-intel — APPROVED** [review]

**Pull request merged**